modules/distributor/distributor.go

```diff
@@ -388,7 +388,9 @@ func (d *Distributor) sendToIngestersViaBytes(ctx context.Context, userID string
 
 	var mu sync.Mutex
 
-	err := ring.DoBatch(ctx, op, d.ingestersRing, keys, func(ingester ring.InstanceDesc, indexes []int) error {
+	writeRing := d.ingestersRing.ShuffleShard(userID, d.overrides.IngestionTenantShardSize(userID))
```

Comment:
Yup, if shard size is 0 it just returns the ring, effectively a no-op.

https://github.com/grafana/dskit/blob/7bd314b462b234ffdcad12275efa6bcdd63d5be5/ring/ring.go#L665-L668

So it's safe to roll out this code as is. Setting the shard size should only be done if we have set `querier_shuffle_sharding_ingesters_enabled: false` though since lowering shard size means queriers will miss ingesters with data.